domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsControllerTest.kt

```diff
@@ -0,0 +1,496 @@
+import org.oppia.android.app.model.OppiaMetricLog.Priority.LOW_PRIORITY
+import org.oppia.android.app.model.OppiaMetricLog.Priority.MEDIUM_PRIORITY
+import org.oppia.android.domain.oppialogger.EventLogStorageCacheSize
+import org.oppia.android.domain.oppialogger.LoggingIdentifierModule
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.domain.oppialogger.PerformanceMetricsLogStorageCacheSize
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.platformparameter.PlatformParameterSingletonModule
+import org.oppia.android.testing.FakePerformanceMetricsEventLogger
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.data.DataProviderTestMonitor
+import org.oppia.android.testing.logging.SyncStatusTestModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClock
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.data.DataProviders
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.logging.performancemetrics.MetricLogSchedulerModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtil
+import org.oppia.android.util.networking.NetworkConnectionUtil.ProdConnectionStatus.NONE
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TEST_TIMESTAMP = Long.MAX_VALUE
+private const val TEST_CPU_USAGE = Long.MAX_VALUE
+private const val TEST_APK_SIZE = Long.MAX_VALUE
+private const val TEST_STORAGE_USAGE = Long.MAX_VALUE
+private const val TEST_STARTUP_LATENCY = Long.MAX_VALUE
+private const val TEST_NETWORK_USAGE = Long.MAX_VALUE
+private const val TEST_MEMORY_USAGE = Long.MAX_VALUE
+
+/** Tests for [PerformanceMetricsControllerTest]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PerformanceMetricsControllerTest.TestApplication::class)
+class PerformanceMetricsControllerTest {
+
+  @Inject
+  lateinit var performanceMetricsController: PerformanceMetricsController
+
+  @Inject
+  lateinit var oppiaLogger: OppiaLogger
+
+  @Inject
+  lateinit var performanceMetricsLogger: PerformanceMetricsLogger
```

Comment:
This seems like a backward dependency--the logger depends on the controller, not the other way around. I'd expect the controller to be the only API interacted with when verifying its behavior in this test suite (other than supporting APIs).

Reply:
it was an unused variable, removed now.